### ISSUE TITLE
Use rates from cache when currencylayer api throws errors

### ISSUE THREAD

**Issue by donotresist** [phlegx/money-currencylayer-bank#11]
I guess this is kind of a feature request.

Currently, in the gem, if we encounter any of `CurrencyLayer`'s [API Error Codes](https://currencylayer.com/documentation#error_codes) when querying for rate, it raises an error.

https://github.com/phlegx/money-currencylayer-bank/blob/472b553e81ecdfa935a4bc76f818d985c2471e93/lib/money/bank/currencylayer_bank.rb#L278-L280

I can catch this error in my application sure but I don't know if there is a lot of option to handle it gracefully. Say, my monthly usage limit has been reached and currency layer throws that error - I cannot make a new a request (given), but I also don't have any option to use the rates from the cache. And since we are dealing with currency here, depending upon the business logic there will be a lot of things to consider and I cannot just return _anything_ after catching the error. The ideal solution, I think is to return rates from the cache.

As a temporary workaround to this, I am overriding the `exchange_rates` method in my application to be something like this

```
def exchange_rates(straight = false)
    rates = if straight
              raw_rates_straight
            else
              raw_rates_careful
            end
    if rates.key?('quotes')
      @rates = rates['quotes']
    elsif rates.key?('error')
      ### CUSTOM CODE - BEGIN (try returning directly from cache instead of raising Error)
      Rails.logger.warn("CurrencyLayer Error: #{rates.dig('error', 'info')}")
      rates = raw_rates_careful(false)
      @rates = rates.key?('quotes') ? rates['quotes'] : {} 
      ### CUSTOM CODE - END
      
      # ORIGINAL CODE
      # raise Error, rates.dig('error', 'info') 
    else
      raise Error, 'Unknown rates situation!'
    end
  end
```

But I think it would be a good idea, if the gem handled this itself, by giving the dev an option to read rates from cache when this happens (with a flag passed via the opts hash or some other more robust way).

**Comment by synth** [comment]
This happened to us when recently CurrencyLayer didn't go down, per se, but returned empty responses. So, the request looked successful and no error was raised until our application tried to parse the data and then crashed. Ideally, as the OP has said, we can catch any exception on the application side, and return values from cache.

**Comment by phlegx** [comment]
**Released v0.7.0** 🎉
```ruby
# (optional)
# Rescue with rates from the cache instead of reporting an error when the endpoint fails.
mclb.rescue_with_cache = true
```
